hash2curve/src/hash2field/expand_msg/xof.rs

```diff
@@ -66,16 +62,27 @@ where
         domain.update_hash(&mut reader);
         reader.update(&[domain.len()]);
         let reader = reader.finalize_xof();
-        Ok(Self { reader })
+        Ok(Self {
+            reader,
+            length: len_in_bytes,
+        })
     }
 }
 
-impl<HashT> Expander for ExpandMsgXof<HashT>
+impl<HashT> Iterator for ExpandMsgXof<HashT>
 where
     HashT: Default + ExtendableOutput + Update + HashMarker,
 {
-    fn fill_bytes(&mut self, okm: &mut [u8]) {
-        self.reader.read(okm);
+    type Item = u8;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.length == 0 {
+            return None;
+        }
+        self.length -= 1;
+        let mut buf = [0u8; 1];
+        self.reader.read(&mut buf);
+        Some(buf[0])
```

Comment:
See [`array::from_mut()`](https://doc.rust-lang.org/nightly/core/array/fn.from_mut.html) to get rid of the array.
```suggestion
        let mut byte = 0;
        self.reader.read(array::from_mut(&mut byte));
        Some(byte)
```